src/parser.rs

```diff
@@ -50,23 +50,22 @@ fn parse_union_indices(matcher_rule: pest::iterators::Pair<Rule>) -> Vec<UnionEl
 
     for r in matcher_rule.into_inner() {
         match r.as_rule() {
-            Rule::unionChild => res.append(&mut parse_union_child(r)),
+            Rule::unionChild => res.push(parse_union_child(r)),
```

Comment:
Now that both are pushes, can we just implement this with a map + collect?

Reply:
Yup. Well spotted. Fixed.